src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperUseageDescriptorFactory.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if !DNXCORE50 // Cannot accurately resolve the location of the documentation XML file in coreclr.
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNet.Razor.TagHelpers;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    public static class TagHelperUseageDescriptorFactory
+    {
+        private static readonly Dictionary<string, XmlDocumentationProvider> XmlDocumentationProviderCache =
+            new Dictionary<string, XmlDocumentationProvider>(StringComparer.Ordinal);
+        private static readonly char[] InvalidPathCharacters = Path.GetInvalidPathChars();
+
+        public static TagHelperUseageDescriptor CreateDescriptor([NotNull] TypeInfo typeInfo)
+        {
+            var id = XmlDocumentationProvider.GetId(typeInfo);
+            return CreateDescriptorCore(typeInfo.Assembly, id);
+        }
+
+        public static TagHelperUseageDescriptor CreateDescriptor([NotNull] PropertyInfo propertyInfo)
+        {
+            var id = XmlDocumentationProvider.GetId(propertyInfo);
+            var declaringTypeAssembly = propertyInfo.DeclaringType.GetTypeInfo().Assembly;
```

Comment:
Throughout this PR, remove `Type` from `TypeAssembly` names. Every assembly contains types. `declaringAssembly` here, `assemblyLocation` (or just `location`) a few lines down, and so on.